interfaces/acados_template/acados_template/acados_casadi_ocp_solver.py

```diff
@@ -964,3 +965,191 @@ def cost_get(self, stage_: int, field_: str) -> np.ndarray:
+        """
+        if self.nlp_sol is None:
+            raise ValueError('No solution available. Please call solve() first.')
+
+        _, lambda_value, _, _ = self.get_constraints_value(stage)
+        _, _, _, active_ineq_lb_indices, active_ineq_ub_indices = self.get_constraints_indices(stage)
+
+        for i in active_ineq_lb_indices:
+            lam = np.maximum(0, -lambda_value[i])
+            if lam < tol:
+                return False
+        for i in active_ineq_ub_indices:
+            lam = np.maximum(0, lambda_value[i])
+            if lam < tol:
+                return False
+        return True
+
+    def check_strict_complementarity(self, tol: float = 1e-6) -> bool:
+        """
+        Check if the solution satisfies strict complementarity conditions for all stages.
+        Not tested yet.
+        """
+        if self.nlp_sol is None:
+            raise ValueError('No solution available. Please call solve() first.')
+        dims = self.ocp.dims
+        complementarity = []
+
+        for stage in range(dims.N + 1):
+                complementarity.append(self.check_strict_complementarity_stage(stage, tol))
+        return complementarity
+
+    def check_LICQ_stage(self, stage):
+        """
+        Check if the solution satisfies the Linear Independence Constraint Qualification (LICQ) for a given stage.
+        """
+        if self.nlp_sol is None:
+            raise ValueError('No solution available. Please call solve() first.')
+
+        _, eq_indices_x, eq_indices_constraints, active_ineq_lb_indices, active_ineq_ub_indices = self.get_constraints_indices(stage)
+
+        w, w_value, constraints_expr_stage, eq_indices = self._get_w_and_constraints_for_LICQ(stage, eq_indices_x, eq_indices_constraints)
+
+        eq_constraints = constraints_expr_stage[eq_indices] if len(eq_indices) != 0 else ca.vertcat()
+        active_ineq_lb_constraints = constraints_expr_stage[active_ineq_lb_indices] if len(active_ineq_lb_indices) != 0 else ca.vertcat()
+        active_ineq_ub_constraints = constraints_expr_stage[active_ineq_ub_indices] if len(active_ineq_ub_indices) != 0 else ca.vertcat()
+        constraint_matrix = ca.vertcat(eq_constraints,
+                                       -active_ineq_lb_constraints,
+                                       active_ineq_ub_constraints)
+        constraint_jac_expr = ca.Function('constraint_jac', [w], [ca.jacobian(constraint_matrix, w)])
+        constraint_jac_value = constraint_jac_expr(w_value).full()
+
+        # Check if the Jacobian of the constraints is full rank
+        rank = np.linalg.matrix_rank(constraint_jac_value) if constraint_jac_value.any() else 0
+        if rank == constraint_jac_value.shape[0]:
+            return True
+        else:
+            return False
+
+    def check_LICQ(self, tol: float = 1e-6) -> bool:
+        """
+        Check if the solution satisfies the Linear Independence Constraint Qualification (LICQ) for all stages.
+        """
```

Comment:
I would rename to `satisfies_LICQ`, and only return a bool, as indicated in the typehint.
You can introduce an intermediate function that returns the stage-wise array, which might be useful to investigate if the check fails.